### DISCUSSION TITLE
Replace Datadog's zstd with Klauspost's zstd

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#2618] [Ideas]
**What would you like to be added**:

Replace datadog's zstd with Klauspost's zstd

https://github.com/DataDog/zstd
https://github.com/klauspost/compress

**Why is this needed**:

Klauspost's package are pure Go and without use of "unsafe", highly maintained and optimized and also provide support for other compression methods. BadgerDB made the same move: https://github.com/dgraph-io/badger/pull/1709

**Comment by davies** [comment]
The original Zstd implementation is more battle tested and performant, we will NOT replace it with a pure Go version in near future, because we compress/decompress the block in one shot, the CGO is not the problem here.

In the case that when CGO is not available, we may switch to the pure Go version using build tags.

**Reply by solracsf** [reply]
Got it, thanks.
